core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -767,6 +767,51 @@ private void assignFilters(QueryModel parent) throws SqlException {
         assert postFilterRemoved.size() == pc;
     }
 
+    private void authorizeColumnAccess(SqlExecutionContext executionContext, QueryModel model) {
+        if (model != null) {
+            if (model.getTableName() != null) {
+                ExpressionNode tableNameExpr = model.getTableNameExpr();
+                switch (tableNameExpr.type) {
+                    case CONSTANT:
+                    case LITERAL:
+                        // adjust name in case of marker
+                        GenericLexer.FloatingSequence tab = (GenericLexer.FloatingSequence) tableNameExpr.token;
+                        int lo = tab.getLo();
+                        if (Chars.startsWith(tab, QueryModel.NO_ROWID_MARKER)) {
+                            tab.setLo(lo + QueryModel.NO_ROWID_MARKER.length());
+                        }
+                        try {
+                            // copy column names
+                            literalCollectorANames.clear();
+                            final ObjList<QueryColumn> topDownColumns = model.getTopDownColumns();
+                            for (int i = 0, n = topDownColumns.size(); i < n; i++) {
+                                literalCollectorANames.add(topDownColumns.getQuick(i).getName());
+                            }
+
+                            executionContext.getSecurityContext().authorizeSelect(
+                                    executionContext.getTableToken(tab),
+                                    literalCollectorANames
+                            );
+                        } finally {
+                            tab.setLo(lo);
+                        }
+                        break;
+                    default:
+                        // todo: function call
```

Comment:
nit: do we have any plans to secure function calls?

Reply:
yeah, this is also a WIP. Some function might divulge privileged information. We need to secure function calls.

Reply:
Marking this one as resolved as it's something to be addressed in a future PR.